compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala

```diff
@@ -1038,4 +1038,26 @@ class ErrorMessagesTests extends ErrorMessagesTest {
         val ExpectedTypeBoundOrEquals(found) :: Nil = messages
         assertEquals(Tokens.IDENTIFIER, found)
       }
+
+  @Test def classAndCompanionNameClash =
+    checkMessagesAfter("refchecks") {
+      """
+        |class T {
+        |  trait G
```

Comment:
>I would make it a trait, to make the test case more interesting

Sorry if it wasn't clear, this one should be `class G`. I only meant to change the one in `object T`:
```scala
class T {
  class G
}
object T {
  trait G
}
```

Reply:
:+1: